src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorHtmlPublishDiagnosticsInterceptor.cs

```diff
@@ -59,14 +60,21 @@ public override async Task<InterceptionResult> ApplyChangesAsync(JToken token, s
             }
 
             cancellationToken.ThrowIfCancellationRequested();
+
+            await InitializeLogHubLoggerAsync().ConfigureAwait(false);
```

Comment:
Could you clarify why we need to "initialize" in the diagnostics interceptor?

Reply:
The diagnostics interceptor isn't a part of our normal Razor HTML CSharp LS stack in the sense it's lifecycle is a bit different. It initializes before the actual language server, as we export it to be used by WTE directly. 

Consequently, if we don't initialize the logger here, then the logger will be unavailable for logging.

Reply:
Ahhh fair point, I wonder if this would benefit from a general ask that @alexgav had on how we can validate that sub-language servers have been initialized.

At the very least I'd add a comment explaining